hammerblade/torch/kernel/hb_tiled_for.hpp

```diff
@@ -260,13 +345,25 @@ inline void hb_tiled_foreach(HBTensor<scalar_t> res,
     strides[2] = (tensor1.get_strides())[0];
     strides[3] = (tensor2.get_strides())[0];
```

Comment:
We don't have to use byte strides since we only support `uint32_t` and `float`. Having to support all possible types like `uint8_t`, `uint16_t` and `uint64_t` unnecessarily adds multiplies during indexing, and even if we do that there are other parts of hb-pytorch backend that prevents us from using this feature anyway! But this a separate idea we can discuss in an issue probably.

Reply:
I can totally see what you are saying ;)

yeah the problem of TensorIterator is ... it's just complicated. Just realized that it actually reorder your dimensions. Like a(x, y, z), normally x is index0 and y is index1. But TensorIterator reverse it. It makes z index0 instead of index2.

Reply:
But byte strides would still be in the last dimension?

Reply:
Emmm I guess my point is, even without this strides in bytes thing, we still need to distinguish between "regular tensor" and "iterator tensor", because of the reordering. Well technically it won't be a correctness issue, I guess. Basically that means we treat a (a, b, c) tensor as a (c, b, a) tensor. As long as the strides and sizes are correct (they are reordered as well), we have correct results.

I mean I have no strong objection of removing the bytes ;)